app/src/main/java/org/mozilla/fenix/settings/address/view/AddressEditorView.kt

```diff
@@ -160,9 +202,107 @@ class AddressEditorView(
 
         binding.countryDropDown.adapter = adapter
         binding.countryDropDown.setSelection(selectedPosition)
+        binding.countryDropDown.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                parent: AdapterView<*>?,
+                view: View?,
+                position: Int,
+                id: Long
+            ) {
+                val newCountryKey = binding.countryDropDown.selectedItem.toString().toCountryCode()
+                countries[newCountryKey]?.let { country ->
+                    bindSubregionDropdown(country)
+                }
+            }
+
+            override fun onNothingSelected(p0: AdapterView<*>?) = Unit
+        }
+        return countries[selectedCountryKey]
     }
 
     private fun String.toCountryCode() = countries.values.find {
         it.displayName == this
     }?.key ?: DEFAULT_COUNTRY
+
+    companion object {
+        private val canadianSubregions = listOf(
```

Comment:
Can you add a reference to where you got the values of this list from in the code